backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/businessentities/VdsCpuUnit.java

```diff
@@ -139,7 +139,8 @@ public boolean pinVm(Guid vmId, CpuPinningPolicy cpuPinningPolicy) {
         if (!this.vmIds.contains(vmId)) {
             this.vmIds.add(vmId);
         }
-        if (cpuPinningPolicy == CpuPinningPolicy.RESIZE_AND_PIN_NUMA) {
+        // CpuPinningPolicy.NONE may happen when the engine generates CPU pinning based on the NUMA pinning.
+        if (cpuPinningPolicy == CpuPinningPolicy.RESIZE_AND_PIN_NUMA || cpuPinningPolicy == CpuPinningPolicy.NONE) {
```

Comment:
had we set it usinga boolean value of exclusiveness, we could maybe avoid this code (assuming that components that examine the topology shouldn't care about why the other vms took those cpus, i.e., what policy lead to that, but only if a given cpu is exclusively pinned, not-exclusively pinned or not pinned at all) - but let's not change it now